kwc-button.html

```diff
@@ -0,0 +1,378 @@
+                    on-keys-pressed="_keysPressed">
+                    </iron-a11y-keys>
+    <template is="dom-if" if="[[_displayIcon]]">
+        <iron-icon icon="kano-icons:[[icon]]"></iron-icon>
+    </template>
+    <slot id="content"></slot>
+  </template>
+
+  <script>
+    Polymer({
+      is: 'kwc-button',
+      properties: {
+            /**
+             * Boolean to indicate whether the button is `active` or `disabled`
+             */
+            disabled: {
+                type: Boolean,
+                value: false,
+                reflectToAttribute: true
+            },
+            _displayIcon: {
+                type: Boolean,
+                computed: '_iconProvided(icon)'
+            },
+            /**
+             * Boolean to indicate whether the button ghosted (semi-transparent)
+             */
+            ghost: {
+                type: Boolean,
+                value: false,
+                reflectToAttribute: true
+            },
+            /**
+             * String to indicate which button (if any) the button
+             * should display
+             */
+            icon: {
+                type: String
+            },
+            /**
+             * Boolean to indicate whether the button should only have
+             * an outline
+             */
+            outline: {
+                type: Boolean,
+                value: false,
+                reflectToAttribute: true
+            },
+            /**
+             * Boolean to indicate whether the button should be square
+             */
+            square: {
+                type: Boolean,
+                value: false,
+                reflectToAttribute: true
+            },
+            _target: {
+                type: Object,
+                value: function () {
+                    return this;
+                }
+            },
+            /**
+             * Boolean to indicate whether the button should be
+             * completely transparent
+             */
+            transparent: {
+                type: Boolean,
+                value: false,
+                reflectToAttribute: true
+            },
+            /**
+             * String to indicate the role played by the button, and the styling
+             * it should use:
+             *    - `primary`
+             *    - `secondary`
+             *    - `tertiary`
+             *    - `warning`
+             *    - `white`
+             * Named `type` to avoid clashes with the HTML `role` attribute
+             */
+            type: {
+                type: String,
+                value: 'primary',
+                reflectToAttribute: true
+            }
+        },
+        hostAttributes: {
+            role: 'button',
+            tabindex: '0'
+        },
+        _iconProvided (icon) {
+            return icon !== undefined && icon !== null;
+        },
+        _keysPressed (update) {
+            this.fire('click');
```

Comment:
Firing generic events like this is controversial in my opinion. That means that any parent will be able to catch this. Also if any parent prevent defaults it can prevent the desired parent to catch it. Either way I felt a bit worried about it before and I still am.

Reply:
Yep, sorry, I meant to remove this for the moment and do some more research into how to handle this better.